### PR TITLE
feat: adds ldap-escape

### PR DESCRIPTION
**Pull request by simonecorsi**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/ldap-escape/ldap-escape-tests.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
+/* tslint:disable:no-namespace */
+'use strict';
```
Comment (gabritto) [inline types/ldap-escape/ldap-escape-tests.ts:2]: I don't think you need `'use strict'` here.

File: types/ldap-escape/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+// Type definitions for ldap-escape 2.0
+// Project: https://github.com/tcort/ldap-escape
+// Definitions by: Simone Corsi <https://github.com/simonecorsi>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export function filter(TemplateStringsArray: TemplateStringsArray, ...str: string[] | number[]): string;
+export function dn(TemplateStringsArray: TemplateStringsArray, ...str: string[] | number[]): string;
+
+export as namespace LdapEscape;
```
Comment (gabritto) [inline types/ldap-escape/index.d.ts:9]: Why do we need this `export as namespace`? Looking at the original code, it seems to me the original library only has named exports, so I think the right thing to do is to not have the `export as namespace`, unless I'm missing something.

File: types/ldap-escape/ldap-escape-tests.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import { dn, filter } from 'ldap-escape';
+
+namespace Module {
```
Comment (gabritto) [inline types/ldap-escape/ldap-escape-tests.ts:6]: I also don't think you need this namespace here, or the `/*tslint:disable:no-namespace */` line if you get rid of the namespace.

**Review by gabritto — CHANGES_REQUESTED** [review]

File: types/ldap-escape/index.d.ts (hunk 2 above)
Comment (simonecorsi) [reply types/ldap-escape/index.d.ts:9]: You're right, that was a biased change I didn't think enough before doing, I'm gonna change it!

**Comment by simonecorsi** [comment]
@gabritto do you need me to squash the commits after the review?

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@simonecorsi thanks!

**Comment by simonecorsi** [comment]
Ready to merge

**Pull request merged**